pkg/rdkafka/RdKafkaContext.php

```diff
@@ -139,6 +139,12 @@ public function close(): void
         foreach ($kafkaConsumers as $kafkaConsumer) {
             $kafkaConsumer->unsubscribe();
         }
+
+        // Compatibility with phprdkafka 4.0.
+        if (isset($this->producer) && method_exists($this->producer, 'flush')) {
+            $this->producer->flush($this->config['shutdown_timeout'] ?? -1);
```

Comment:
maybe throw an error or log something if the return value is not `RD_KAFKA_RESP_ERR_NO_ERROR`, sry not sure what the enqueue convention is for something like that

Reply:
As what we discussed: throwing an exception here would change the behavior and make it necessary for wrapping libraries (like enqueue-bundle) to catch it, so I've left it out.

However, I do agree that it is something that this package eventually should do, since otherwise potential delivery errors will be hidden from end user.